homeassistant/components/homeassistant_hardware/coordinator.py

```diff
@@ -0,0 +1,85 @@
+"""Home Assistant hardware firmware update coordinator."""
+
+from __future__ import annotations
+
+from datetime import timedelta
+import logging
+
+from aiohttp import ClientSession
+from yarl import URL
+
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .models import FirmwareManifest
+
+_LOGGER = logging.getLogger(__name__)
+
+
+FIRMWARE_REFRESH_INTERVAL = timedelta(hours=8)
+NABU_CASA_FIRMWARE_RELEASES_URL = (
+    "https://api.github.com/repos/NabuCasa/silabs-firmware-builder/releases/latest"
+)
+
+
+class FirmwareUpdateCoordinator(DataUpdateCoordinator[FirmwareManifest]):
+    """Coordinator to manage firmware updates."""
+
+    def __init__(self, hass: HomeAssistant, session: ClientSession) -> None:
+        """Initialize the firmware update coordinator."""
+        super().__init__(
+            hass,
+            _LOGGER,
+            name="firmware update coordinator",
+            update_interval=FIRMWARE_REFRESH_INTERVAL,
+            always_update=False,
+        )
+        self.hass = hass
+        self.session = session
+
+        self._latest_release_url: str | None = None
+        self._latest_manifest: FirmwareManifest | None = None
+
+    async def _async_update_data(self) -> FirmwareManifest:
+        # Fetch the latest release metadata
+        async with self.session.get(
+            NABU_CASA_FIRMWARE_RELEASES_URL,
```

Comment:
We need a Python client library, published on PyPI, for the requests and the models. We can either host it in the silabs firmware builder repo or in a new repo under the home-assistant-libs org.